src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs

```diff
@@ -20,67 +20,57 @@ public class ActivitySourceTests : IDisposable
+                using ActivitySource as1 = new ActivitySource("Source1");
+                Assert.Equal("Source1", as1.Name);
+                Assert.Equal(String.Empty, as1.Version);
+                Assert.False(as1.HasListeners());
+
+                using ActivitySource as2 =  new ActivitySource("Source2", "1.1.1.2");
+                Assert.Equal("Source2", as2.Name);
+                Assert.Equal("1.1.1.2", as2.Version);
+                Assert.False(as2.HasListeners());
             }).Dispose();
         }
 
         [Fact]
         public void TestStartActivityWithNoListener()
         {
             RemoteExecutor.Invoke(() => {
-                using (ActivitySource aSource =  new ActivitySource("SourceActivity"))
-                {
-                    Assert.Equal("SourceActivity", aSource.Name);
-                    Assert.Equal(string.Empty, aSource.Version);
-                    Assert.False(aSource.HasListeners());
+                using ActivitySource aSource =  new ActivitySource("SourceActivity");
+                Assert.Equal("SourceActivity", aSource.Name);
+                Assert.Equal(string.Empty, aSource.Version);
+                Assert.False(aSource.HasListeners());
 
-                    Activity current = Activity.Current;
-                    using (Activity a1 = aSource.StartActivity("a1"))
-                    {
-                        // no listeners, we should get null activity.
-                        Assert.Null(a1);
-                        Assert.Equal(Activity.Current, current);
-                    }
-                }
+                Activity current = Activity.Current;
+
+                using Activity a1 = aSource.StartActivity("a1");
+                // no listeners, we should get null activity.
+                Assert.Null(a1);
+                Assert.Equal(Activity.Current, current);
             }).Dispose();
         }
 
         [Fact]
         public void TestActivityWithListenerNoActivityCreate()
         {
             RemoteExecutor.Invoke(() => {
-                using (ActivitySource aSource =  new ActivitySource("SourceActivityListener"))
+                using ActivitySource aSource =  new ActivitySource("SourceActivityListener");
+                Assert.False(aSource.HasListeners());
+
+                using ActivityListener listener = new ActivityListener
                 {
-                    Assert.False(aSource.HasListeners());
+                    ActivityStarted = activity => Assert.NotNull(activity),
+                    ActivityStopped = activity => Assert.NotNull(activity),
+                    ShouldListenTo = (activitySource) => object.ReferenceEquals(aSource, activitySource),
+                    GetRequestedDataUsingParentId = (ref ActivityCreationOptions<string> activityOptions) => ActivityDataRequest.None,
+                    GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> activityOptions) => ActivityDataRequest.None
+                };
```

Comment:
How about:
```C#
using var listener = new ActivityListener();
listener.ActivityStarted = activity => Assert.NotNull(activity);
listener.ActivityStopped = activity => Assert.NotNull(activity);
listener.ShouldListenTo = (activitySource) => object.ReferenceEquals(aSource, activitySource);
listener.GetRequestedDataUsingParentId = (ref ActivityCreationOptions<string> activityOptions) => ActivityDataRequest.None;
listener.GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> activityOptions) => ActivityDataRequest.None;
```
?  I realize what you have is valid.  My problem with it is that when I read it, I see the using and immediately think that everything inside the braces is the body of the using, with the instance getting disposed of at the closing brace.  Obviously that's not the case, but it's where 20 years of reading C# leads my mind :)

Reply:
done. Thanks!

Reply:
@stephentoub please let me know if you have any more comments.